app/src/main/java/com/example/dine_and_donate/EditProfileActivity.java

```diff
@@ -142,6 +159,39 @@ public void onClick(View v) {
                 setState(mEditName, mClearName);
             }
         });
+
+        mChangeProfPic.setOnClickListener(new View.OnClickListener() {
+            final Uri[] downloadUri = new Uri[1];
+
+            @Override
+            public void onClick(View v) {
+                pickFromGallery();
+                if(mSelectedImage != null) {
```

Comment:
this is the same code as EventActivity right? it's best not to duplicate code. Can you create a new class called "UploadUtil" or something, then put a helper method there that both EventActivity and EditProfileActivity can use?